src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -700,33 +697,30 @@ export class ActivityFillGraphQuerier {
         ? fullThread.samples.time[fullThreadSample + 1]
         : sampleTime + interval;
 
-    let cpuDeltaBeforeSample = null;
-    let cpuDeltaAfterSample = null;
-    const { threadCPUDelta } = samples;
-    if (enableCPUUsage && threadCPUDelta) {
-      // It must be non-null because we are checking this in the processing
-      // step and eliminating all the null values.
-      cpuDeltaBeforeSample = ensureExists(threadCPUDelta[sample]);
+    let beforeSampleCpuRatio = 1;
+    let afterSampleCpuRatio = 1;
+    const { threadCPURatio } = samples;
+    if (enableCPUUsage && threadCPURatio) {
+      beforeSampleCpuRatio = threadCPURatio[sample];
       // Use the fullThread here to properly get the next in case zoomed in.
-      cpuDeltaAfterSample = ensureExists(fullThread.samples.threadCPUDelta)[
-        fullThreadSample + 1
-      ];
-      cpuDeltaAfterSample =
-        // It can be undefined if this is the last sample in the profile.
-        cpuDeltaAfterSample !== undefined && cpuDeltaAfterSample !== null
-          ? cpuDeltaAfterSample
-          : cpuDeltaBeforeSample;
+      const fullThreadSamplesCPURatio = ensureExists(
+        fullThread.samples.threadCPURatio
+      );
+      if (fullThreadSample + 1 < fullThreadSamplesCPURatio.length) {
+        afterSampleCpuRatio = fullThreadSamplesCPURatio[fullThreadSample + 1];
+      }
```

Comment:
For the last sample, `afterSampleCpuRatio` will be always `1`, right? Is this correct or do we maybe want to use the same value as `beforeSampleCpuRatio`?

Reply:
I'll do what you suggested. I don't think there's a clear answer because we just don't have the information about what happens after the last sample.

Reply:
I think we set the "after last" duration to the interval, so it's true that it would be weird to make it look as if there was 100% CPU for one interval after the last sample. Defaulting to zero would make more sense. But defaulting to `beforeSampleCpuRatio` also sounds fine.

Reply:
Oh, using `beforeSampleCpuRatio` is also what we do in `_accumulateSampleCategories`, so yes this would make hit testing consistent with rendering.